src/Components/ABDM/ABDMRecordsTab.tsx

```diff
@@ -0,0 +1,198 @@
+      className="w-full cursor-pointer overflow-hidden bg-white shadow sm:rounded-lg"
+    >
+      <div className="flex flex-col items-center justify-between gap-4 px-4 py-5 sm:flex-row sm:gap-0 sm:px-6">
+        <div className="flex flex-col items-center">
+          <h5 className="font-semibold leading-6 text-gray-900">
+            {artefact.hip}
+          </h5>
+          <p className="mt-1 max-w-2xl text-sm text-gray-500">
+            created {dayjs(artefact.created_date).fromNow()}
+          </p>
+        </div>
+        <div className="flex flex-col items-center">
+          <h6 className="mt-1 leading-6 text-gray-700">{artefact.status}</h6>
+          <h6 className="mt-1 leading-6 text-gray-700">
+            {dayjs(artefact.from_time).format("MMM DD YYYY")} -{" "}
+            {dayjs(artefact.to_time).format("MMM DD YYYY")}
+          </h6>
+          <p className="mt-1 max-w-2xl text-sm text-gray-500">
+            expires in {dayjs(artefact.expiry).fromNow()}
+          </p>
+        </div>
+      </div>
+      <div className="flex flex-wrap items-center justify-center border-t border-gray-200 px-4 py-5 sm:gap-4">
+        {artefact.hi_types.map((hiType) => {
+          return (
+            <div
+              key={hiType}
+              className="flex items-center justify-center rounded-full bg-gray-600 px-4 py-1.5 text-xs font-medium text-white"
+            >
+              {hiType}
+            </div>
+          );
+        })}
+      </div>
+    </Link>
+  );
+}
+
+interface IConsentRequestCardProps {
+  consent: ConsentRequestModel;
+}
+
+function ConsentRequestCard({ consent }: IConsentRequestCardProps) {
+  const dispatch = useDispatch<any>();
+
+  return (
+    <div className="overflow-hidden bg-white shadow sm:rounded-lg">
+      <div className="flex flex-col items-center justify-between gap-4 px-4 py-5 sm:flex-row sm:gap-0 sm:px-6">
+        <div className="flex flex-col items-center">
+          <h5 className="font-semibold leading-6 text-gray-900">
+            {
+              ABDM_CONSENT_PURPOSE.find((p) => p.value === consent.purpose)
+                ?.label
+            }
+          </h5>
+          <h6 className="mt-1 leading-6 text-gray-700">
+            {consent.requester.first_name} {consent.requester.last_name}
+          </h6>
+        </div>
+        <div className="flex flex-col items-center">
+          <h6 className="mt-1 leading-6 text-gray-700">
+            {dayjs(consent.from_time).format("MMM DD YYYY")} -{" "}
+            {dayjs(consent.to_time).format("MMM DD YYYY")}
+          </h6>
+          <p className="mt-1 max-w-2xl text-sm text-gray-500">
+            expires in {dayjs(consent.expiry).fromNow()}
+          </p>
+        </div>
+        <div className="flex flex-col items-center">
+          <ButtonV2
+            onClick={async () => {
+              const res = await dispatch(routes.abha.checkConsentStatus, {
+                pathParams: { id: consent.id },
+              });
```

Comment:
does this work? useDispatch's dispatch expects a redux action and not a schema definition of the API route right?

Consider switching to `request` instead?